src/test/java/org/springframework/samples/petclinic/cache/CacheEffectivenessTests.java

```diff
@@ -0,0 +1,843 @@
+
+    private String baseUrl;
+    private String apiBaseUrl;
+    private Map<String, EntityHandler> entityHandlers;
+
+    @BeforeEach
+    void setUp() {
+        baseUrl = "http://localhost:" + port + "/petclinic/api/cache";
+        apiBaseUrl = "http://localhost:" + port + "/petclinic/api";
+
+        entityHandlers = Map.of(
+            "Vet", new GenericEntityHandler<>(
+                "Vet", "/vets", vetRepository, VetRepository.class, Vet.class,
+                this::createVet, this::createVetDto, new VetRepositoryOperations()
+            ),
+            "PetType", new GenericEntityHandler<>(
+                "PetType", "/pettypes", petTypeRepository, PetTypeRepository.class, PetType.class,
+                this::createPetType, this::createPetTypeDto, new PetTypeRepositoryOperations()
+            ),
+            "Owner", new GenericEntityHandler<>(
+                "Owner", "/owners", ownerRepository, OwnerRepository.class, Owner.class,
+                this::createOwner, this::createOwnerDto, new OwnerRepositoryOperations()
+            ),
+            "Specialty", new GenericEntityHandler<>(
+                "Specialty", "/specialties", specialtyRepository, SpecialtyRepository.class, Specialty.class,
+                this::createSpecialty, this::createSpecialtyDto, new SpecialtyRepositoryOperations()
+            ),
+            "Visit", new GenericEntityHandler<>(
+                "Visit", "/visits", visitRepository, VisitRepository.class, Visit.class,
+                this::createVisit, this::createVisitDto, new VisitRepositoryOperations()
+            ),
+            "Pet", new GenericEntityHandler<>(
+                "Pet", "/pets", petRepository, PetRepository.class, Pet.class,
+                this::createPet, this::createPetDto, new PetRepositoryOperations()
+            )
+        );
+    }
+
+    @ParameterizedTest(name = "Cache Effectiveness Test for {0}")
+    @MethodSource("getEntityHandlers")
+    void testCacheEffectiveness(String entityName) {
+        EntityHandler handler = entityHandlers.get(entityName);
+        handler.setupEffectivenessMocks();
+        EntityCacheTestConfiguration<Object, Object> config = handler.createConfiguration();
+
+        // First call - should hit repository
+        ResponseEntity<Object[]> firstResponse = restTemplate.getForEntity(
+            apiBaseUrl + config.getApiEndpoint(), Object[].class);
+        assertEquals(HttpStatus.OK, firstResponse.getStatusCode());
+        assertNotNull(firstResponse.getBody());
+
+        // Second call - should use cache
+        ResponseEntity<Object[]> secondResponse = restTemplate.getForEntity(
+            apiBaseUrl + config.getApiEndpoint(), Object[].class);
+        assertEquals(HttpStatus.OK, secondResponse.getStatusCode());
+        assertNotNull(secondResponse.getBody());
+
+        // Verify cache statistics
+        ResponseEntity<Map<String, CacheStatistics>> response = restTemplate.exchange(
```

Comment:
We should not use CacheStatistics because it was not defined in spec.
Lets parse json here.